docs/X_layered_XCP.adoc

```diff
@@ -0,0 +1,109 @@
+
+{empty} +
+{empty}
+
+Copyright (C) 2008-2011 MODELISAR Consortium and 2012-2020 The Modelica Association Project FMI.
+
+This document is licensed under the Attribution-ShareAlike 4.0 International license.
+The code is released under the 2-Clause BSD License.
+The licenses text can be found in the https://raw.githubusercontent.com/modelica/fmi-standard/master/LICENSE.txt[LICENSE.txt] file that accompanies this distribution.
+
+{empty}
+
+== Introduction
+
+=== Intend of this Document
+
+FMI 3.0 was extended also with virtual Electronic Control Units (vECUs) in mind.
+Engineers can measure into ECUs using the XCP protocol:
+
+[quote, ASAM e.V., https://www.asam.net/standards/detail/mcd-1-xcp/[ASAM MCD-1 XCP]]
+____
+The ASAM MCD-1 XCP (Universal Measurement and Calibration Protocol) standard defines a bus-independent, master-slave communication protocol to connect ECUs with calibration systems.
+XCP is short for Universal Measurement and Calibration Protocol. The primary purpose of XCP is to adjust internal parameters and acquire the current values of internal variables of an ECU.
+The first letter X in XCP expresses the fact that the protocol is designed for a variety of bus systems.
+The standard consists of a base standard, which describes memory-oriented protocol services without direct dependencies on specific bus systems.
+Several associate standards contain the transport layer definitions for CAN, FlexRay, Ethernet (UDP/IP and TCP/IP), serial links (SPI and SCI) and USB.
+____
+
+This layered standard describes how an FMU advertises its XCP capabilities to importers and MCD tools using the XCP protocol.
+
+=== Rough Outline of the Approach
+
+The XCP description file for the vECU wrapped as FMU will be placed into the `/extra/org.modelica.fmi.layered_XCP` folder.
+This description file follows the https://www.asam.net/standards/detail/mcd-2-mc/[ASAP MCD-2 MC] standard (aka `ASAP2`, also `A2L`) and customarily carries the file extension `.a2l`.
+
+[quote, ASAM e.V., https://www.asam.net/standards/detail/mcd-2-mc/[ASAP MCD-2 MC]]
+____
+ASAM MCD-2 MC (aka ASAP2) defines the description format of the internal ECU variables used in measurement and calibration.
+Measurement & calibration systems (MC-systems) require this description for both the parameterization of scalar constants, curves and maps of the ECU software and for recording the system's response via measurement variables during real-time testing.
+The description contains information about data types, dimensions, record layouts and memory locations of ECU variables.
+The standard also describes how the variable values are converted into human-readable quantities and displayed in an MC-system.
+____
+
+=== Comments about this Approach
+
+_[This standard will have no effect on the FMU interface, nor the C-API behavior._
+_The XCP server inside the FMU will most likely run in its own thread and leave the rest of the operation of the FMU unaffected._
```

Comment:
XCP server -> XCP slave

[The XCP standard uses the term XCP slave for the part implemented in the ECU/FMU.]

Reply:
Done.